### PR TITLE
fix(vcs): Gitlab oauth Application callback URL

### PR DESCRIPTION
**Pull request by fsamin**
Warning: Users , you may need to update VCS your configuration

### CODE REVIEW

File: engine/vcs/vcs.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (s *Service) getConsumer(name string) (sdk.VCSServer, error) {
 		return gitlab.New(serverCfg.Gitlab.AppID,
 			serverCfg.Gitlab.Secret,
 			serverCfg.URL,
-			s.Cfg.API.HTTP.URL+"/repositories_manager/oauth2/callback",
```
Comment (bnjjj) [inline engine/vcs/vcs.go]: Why not keep this value as default if we don't indicate `OAuthCallbackURL` ?

Comment (fsamin) [reply engine/vcs/vcs.go]: Because the other value doesn't have any sens :/

File: engine/vcs/types.go [hunk 2]
```diff
@@ -84,9 +84,10 @@ var errGithubConfigurationError = fmt.Errorf("Github configuration Error")
 
 // GitlabServerConfiguration represents the gitlab configuration
 type GitlabServerConfiguration struct {
-	AppID  string `toml:"appId" json:"-" comment:"#######\n CDS <-> Gitlab. Documentation on https://ovh.github.io/cds/hosting/repositories-manager/gitlab/ \n#######"`
-	Secret string `toml:"secret" json:"-"`
-	Status struct {
+	AppID            string `toml:"appId" json:"-" comment:"#######\n CDS <-> Gitlab. Documentation on https://ovh.github.io/cds/hosting/repositories-manager/gitlab/ \n#######"`
+	Secret           string `toml:"secret" json:"-"`
+	OAuthCallbackURL string `toml:"callbackUrl" json:"OAuth Application Callback URL"`
```
Comment (richardlt) [inline engine/vcs/types.go]: Can you add a default value with http://localhost:8081/repositories_manager/oauth2/callback, to generate default config that works for dev purpose.

Comment (yesnault) [reply engine/vcs/types.go]: json:"OAuth Application Callback URL" --> this is probably the "comment" annotation, not the json annotation :)

Comment (fsamin) [reply engine/vcs/types.go]: fixed

**Pull request merged**